### PR TITLE
codegen: Allow running from anywhere (e.g. workspace root)

### PR DESCRIPTION
**Pull request by MarijnS95**
Depends on #298

When running `cargo r -p codegen` straight from the root, `GLAM_ROOT = ".."` ends up outside the current directory, forcing the user to `cd` into `codegen/` first.  While `GLAM_ROOT` could have been updated with some smart detection `tera` templates rely on the PWD (or should alternatively receive more elaborate paths).  Change into the `codegen/` directory for the user so that they don't have to do this themselves, making the workflow ever so slightly more convenient.

### CODE REVIEW

File: codegen/src/main.rs [hunk 1]
```diff
@@ -517,6 +517,9 @@ fn build_output_pairs() -> HashMap<&'static str, tera::Context> {
 }
 
 fn main() -> anyhow::Result<()> {
+    // Change into `./codegen` dir
+    std::env::set_current_dir(env!("CARGO_MANIFEST_DIR"))?;
```
[the remark this answers is not in the document]
Comment (MarijnS95) [reply codegen/src/main.rs:521]: The environment variable is set by the compiler, so the binary is locked into place. Not something that can be shipped but I don't think that's intended yet?

I proposed an alternative in the PR description, based on how I initially implemented this: look at `current_dir()?.file_name() == "codegen"` to determine `GLAM_ROOT`, but that requires a relative path to be passed into `tera` or `set_current_dir` for the determined directory of `"codegen"`. Do you prefer one of those?

File: codegen/Cargo.toml [hunk 2]
```diff
@@ -6,7 +6,7 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-anyhow = { version = "1", default-features = false }
```
Comment (MarijnS95) [inline codegen/Cargo.toml:9]: @bitshifter I enabled the (default) `"std"` feature here to get `From<>` implementations for any `Error` into an `anyhow::Error`, but it looks like you handled that everywhere else by `match` with `Ok(x) => x` and `Err(e) => bail!("some context {}")` - what is desired?

File: codegen/src/main.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (bitshifter) [reply codegen/src/main.rs:521]: I'm ok with the env var option. I don't expect to ship this tool.

File: codegen/Cargo.toml (hunk 2 above)
Comment (MarijnS95) [reply codegen/Cargo.toml:9]: Fixed in #308 :)

**Pull request merged**